browser/request_otr/request_otr_tab_helper.cc

```diff
@@ -41,19 +44,35 @@ void RequestOTRTabHelper::DidFinishNavigation(
     return;
   }
 
-  RequestOTRStorageTabHelper* storage =
-      RequestOTRStorageTabHelper::GetOrCreate(web_contents());
-  // GetOrCreate should only return null if the runtime flag is disabled, and
+  content::BrowserContext* browser_context =
+      web_contents()->GetBrowserContext();
+  request_otr::RequestOTRService* request_otr_service =
+      request_otr::RequestOTRServiceFactory::GetForBrowserContext(
+          browser_context);
+  // This should only return null if the runtime flag is disabled, and
   // this tab helper should never even be created if the runtime flag is
   // disabled, so a null here is bad news.
-  DCHECK(storage);
-  if (!storage) {
+  DCHECK(request_otr_service);
+  if (!request_otr_service) {
     return;
   }
-  if (!storage->has_requested_otr()) {
+  const GURL& url = navigation_handle->GetURL();
+  if (!request_otr_service->RequestedOTR(url)) {
     return;
   }
 
+  // History service is already patched so request-otr URLs never get added,
```

Comment:
this is new behavior added in this PR that was not part of the previous implementation or it was done somewhere else before?

Reply:
nvm, I see now that several issues were closed by this PR